test/images/agnhost/Dockerfile

```diff
@@ -12,6 +12,23 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# builder image
+# populate the fake registry with images
+ARG GOLANG_VERSION
+FROM golang:$GOLANG_VERSION AS preparer
+
+RUN go install github.com/google/go-containerregistry/cmd/crane@latest && \
+    apt-get update && apt-get install -y jq
+
+COPY fakeregistryserver/prepare_image.sh /prepare_image.sh
+RUN chmod +x /prepare_image.sh
+
+# for now it's only pause:3.10.1 which available internally 
+# as "testing" tag, but we can download more images if needed
+ARG PAUSE_VERSION=3.10.1
+RUN /prepare_image.sh pause $PAUSE_VERSION testing /registry
```

Comment:
how big is this going to be? will it pull in windows stuff? will this work with both arm64 and amd64 boxes?

Reply:
here are the linux sizes, in bytes: https://github.com/kubernetes/kubernetes/issues/130271#issuecomment-3120304752

Reply:
Windows is **much** bigger :/

```
$ crane manifest registry.k8s.io/pause:3.10.1 | jq -r '.manifests[].digest' | while read -r manifest_digest; do crane export registry.k8s.io/pause@${manifest_digest} - | tar -xzf - --to-stdout | wc -c; done
  735760
  345168
  514000
  796712
  553416
 259733994
 297711163
 ```

Reply:
@dims this script pulling everything from https://registry.k8s.io/v2/pause/manifests/3.10.1, which includes linux and windows, and in terms of arch, contains `amd64`, `arm`, `arm64`, `ppc64le` and `s390x`

The full image is 661MB right now in my system

Reply:
For comparison:
```
$ crane export registry.k8s.io/e2e-test-images/agnhost:2.56 - | tar -xzf - --to-stdout | wc -c
 131212932
 ```

or ~131 MB
 
 The windows pause images here are going to add a lot to this image. The linux images are fine.

Reply:
I change the script to filter windows images and the image size become 89.9MB

Reply:
That sounds good, yes 👍 I've added image filtering